File: pset6/readability/readability.py
from cs50 import get_string

def count_letters(text):
    numLet = 0
    for i in text:
        if (i >= 'a' and i <= 'z') or (i >= 'A' and i <= 'z'):
            numLet += 1
    return numLet
    
def count_words(text):
    numWords = 1
    for i in text:
        if i.isspace():
            numWords += 1
    return numWords
    
def count_sentences(text):
    numSent = 0
    for i in text:
        if i == '.' or i == '!' or i == '?':
            numSent += 1
    return numSent
    
def main():
    # get text from user
    text = get_string("Text: ")
    
    # get key info
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    
    # calculate the grade level
    grade = 0.0588 * (100 * letters / words) - 0.296 * (100 * sentences / words) - 15.8
    level = round(grade)
    
    if level >= 16:
        print('Grade 16+')
    elif level < 1:
        print('Before Grade 1')
    else:
        print(f'Grade {level}')
        

main()



    

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void) {
    // get text from user
    string text = get_string("Text: ");
    
    // get key information
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    
    // calculate the grade level
    float grade = 0.0588 * (100 * (float) letters / words) - 0.296 * (100 * (float) sentences / words) - 15.8;
    int level = round(grade);
    
    // determine the grade level
    if(level >= 16) {
        printf("Grade 16+\n");
    } else if(level < 1) {
        printf("Before Grade 1\n");
    } else {
        printf("Grade %i\n", level);
    }
}

// function to count letters
int count_letters(string text) {
    // count the number of letters
    int count = 0;
    for(int i = 0; i < strlen(text); i++) {
        if((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z')) {
            count++;
        }
    }
    // return count
    return count;
}

// function to count words
int count_words(string text) {
    // count the number of words
    int count = 1;
    for(int i = 0; i < strlen(text); i++) {
        if(isspace(text[i])) {
            count++;   
        }
    }
    // return count
    return count;
}

// function to count sentences 
int count_sentences(string text) {
    // count the number of sentences
    int count = 0;
    for(int i = 0; i < strlen(text); i++) {
        if(text[i] == '.' || text[i] == '!' || text[i] == '?') {
            count++;
        }
    }
    // return count
    return count;
}